### PR TITLE
Implements PriorityTaskQueue with std's LinkedList rather than a custom linked list implementation

### PR DESCRIPTION
**Pull request by michaelRichards99**
Fixes https://github.com/hermitcore/libhermit-rs/issues/377.

### CODE REVIEW

**Comment by stlankes** [comment]
Looks good! But I will discuss it with @mkroening Just a double check.

File: src/scheduler/task.rs [hunk 1]
```diff
@@ -356,49 +305,22 @@ impl PriorityTaskQueue {
...
-						self.queues[i].tail = borrow.prev.as_ref().cloned();
-					}
-
-					borrow.prio = prio;
-					borrow.next = None;
-					borrow.prev = None;
-				}
+		for (index, current_task) in self.queues[i].iter().enumerate() {
+			// Move the task from its old list to the new list.
+			if handle.id == current_task.borrow().id {
```
Comment (mkroening) [inline src/scheduler/task.rs:311]: This could be simplified by using `Iterator::find`.

File: src/scheduler/task.rs [hunk 2]
```diff
@@ -356,49 +305,22 @@ impl PriorityTaskQueue {
...
+				// Calling remove is unstable: https://github.com/rust-lang/rust/issues/69210
+				let mut split_list = self.queues[i].split_off(index);
+				let task = split_list.pop_front().ok_or(())?;
+				self.queues[i].append(&mut split_list);
```
Comment (mkroening) [inline src/scheduler/task.rs:315]: We might want to extract this code into a function (be it inline or outside this method) for clarity.

**Review by mkroening — CHANGES_REQUESTED** [review]
This looks good! Thanks a lot!

Two minor remarks, though the current code is definitely an improvement already. 👍

**Comment by michaelRichards99** [comment]
Thank you for the feedback :). We've addressed those 2 changes and made sure the priority bitmap is properly set when removing a task from a list. 

Thank you @ynchen2829 for authoring these changes.

File: src/scheduler/task.rs [hunk 3]
```diff
@@ -272,6 +272,30 @@ impl PriorityTaskQueue {
 		task
 	}
 
+	/// Remove the task at index from the queue and return that task,
+	/// or None if the index is out of range or the list is empty.
+	fn remove_from_queue(
```
Comment (mkroening) [inline src/scheduler/task.rs:277]: Ah, nice! You even made this a fully fledged method. 👍

**Review by mkroening — APPROVED** [review]
Looks good to me!

Thanks a lot! :)

bors r+

**Pull request merged**